benchmarking/runs/concentrate_timeslots/concentrate_timeslot_gender_and_race.py

```diff
@@ -0,0 +1,240 @@
+
+
+def additive_utility_function(student: Student, team: TeamShell) -> float:
+    if len(team.requirements) == 0:
+        return 0.0
+
+    return sum(
+        [student.meets_requirement(requirement) for requirement in team.requirements]
+    ) / float(len(team.requirements))
+
+
+class ConcentrateTimeSlotAndConcentrateGenderRace(Run):
+    @staticmethod
+    def better_algorithm_name(algorithm_name: str) -> str:
+        algorithm_name_dict = {
+            "AlgorithmType.DRR-default": "Double Round Robin",
+            "AlgorithmType.WEIGHT-default": "Weight (Concentrate)",
+            "AlgorithmType.PRIORITY-default": "Priority (Concentrate)",
+            "AlgorithmType.RANDOM-default": "Random",
+            "AlgorithmType.GROUP_MATCHER-default": "Group Matcher",
+        }
+
+        return algorithm_name_dict.get(algorithm_name, algorithm_name)
+
+    @staticmethod
+    def better_metric_name(metric_name: str) -> str:
+        metric_name_dict = {
+            "PrioritySatisfaction": "Priority Satisfaction",
+            "AverageProjectRequirementsCoverage": "Average Project Coverage",
+            "AverageCosineDifference": "Average Intra-Heterogeneity",
+            "AverageSoloStatus": "Average Solo Status",
+            "AverageSoloStatusGender": "Average Solo Status Gender",
+            "AverageTimeslotCoverage": "Average Common Time Availability",
+        }
+
+        return metric_name_dict.get(metric_name, metric_name)
+
+    TEAM_SIZE = 4
+
+    def start(self, num_trials: int = 30, generate_graphs: bool = True):
+        scenario = ConcentrateTimeslotsAndConcentrateGenderAndConcentrateRace(
+            max_num_choices=5
+        )
+
+        metrics = {
+            "PrioritySatisfaction": PrioritySatisfaction(
+                goals_to_priorities(scenario.goals),
+                False,
+            ),
+            "AverageTimeslotCoverage": AverageTimeslotCoverage(
+                available_timeslots=list(range(10)),
+            ),
+            "AverageCosineDifference": AverageCosineDifference(
+                attribute_filter=[
+                    ScenarioAttribute.GENDER.value,
+                    ScenarioAttribute.RACE.value,
+                ],
+            ),
+            "AverageSoloStatus": AverageSoloStatus(
+                minority_groups={
+                    ScenarioAttribute.GENDER.value: [Gender.FEMALE.value],
+                    ScenarioAttribute.RACE.value: [Race.African.value],
+                },
+            ),
+        }
+        simulation_sets = {}
+
+        class_sizes = [20, 100, 240, 500, 1000]
+        # class_sizes = [20, 40, 60, 80, 100]
+
+        for class_size in class_sizes:
+            print("CLASS SIZE /", class_size)
+            student_provider = TimeslotCustomStudentProvider(class_size)
+            cache_key = f"timeslot_stuff/concentrate_timeslots_gender_race/class_size_{class_size}"
+
+            simulation_settings_1 = SimulationSettings(
```

Comment:
wait why are there 3 of these?

Reply:
Oh it was for running on different machine, I'll fix them